packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -145,11 +145,11 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
   }
 
   public componentWillReceiveProps(newProps: IComboBoxProps) {
-    // In controlled component usage where selectedKey is provided, update the selectedIndex
-    // and currentOptions state if the key or options change
-    if (newProps.selectedKey !== undefined &&
+    // In controlled component usage where selectedKey or value is provided,
+    // update the selectedIndex and currentOptions state if the selectedKey or options have changed
+    if ((newProps.selectedKey || newProps.value) &&
       (newProps.selectedKey !== this.props.selectedKey || newProps.options !== this.props.options)) {
-      let index: number = this._getSelectedIndex(newProps.options!, newProps.selectedKey);
+      let index: number = newProps.selectedKey !== undefined ? this._getSelectedIndex(newProps.options!, newProps.selectedKey) : -1;
```

Comment:
can you check explicitly if newProps.options is undefined instead of suppressing it?